restli-server/src/main/java/com/linkedin/restli/internal/server/response/ResponseUtils.java

```diff
@@ -84,6 +95,209 @@ static Object translateCanonicalKeyToAlternativeKeyIfNeeded(Object canonicalKey,
     }
   }
 
+  public static DataMap fillInDefaultOnRecord(RecordDataSchema schema, DataMap dataMap)
+  {
+    DataMap dataWithDefault = new DataMap(DataMapBuilder.getOptimumHashMapCapacityFromSize(dataMap.size()));
+    dataWithDefault.putAll(dataMap);
+    for (RecordDataSchema.Field field : schema.getFields())
+    {
+      DataSchema dataFieldSchema = field.getType();
+      if (dataFieldSchema.getType() == DataSchema.Type.RECORD)
+      {
+        if (dataMap.containsKey(field.getName()) || field.getDefault() != null)
+        {
+          DataMap originalData = dataMap.containsKey(field.getName()) ? (DataMap) dataMap.get(field.getName()) : (DataMap) field.getDefault();
+          dataWithDefault.put(field.getName(), fillInDefaultOnRecord((RecordDataSchema) dataFieldSchema, originalData));
```

Comment:
if you are filling in default datamap you shouldn't need to recursively call for record again as all default would be there from schema

Reply:
Well, according to Karthik's example, even there is a default for a record, we still need to figure out other field to see if there is a default. Let me know if we need a sync with Karthik

Reply:
oh yeah no need thanks